File: d0708/whatever.java
import java.util.*;
class Node {
  Node getNext() {
    return this;
  }
}

public class LoopInspector {

  Map mp;
  LoopInspector() {
    mp = new HashMap<Node, Integer>();
  }
  Integer find(Node nnn, Integer dep) {
    mp.put(nnn, dep);
    if (mp.containsKey(nnn.getNext())) {
      return dep - (Integer)mp.get(nnn.getNext()) + 1;
    }
    return find(nnn.getNext(), dep + 1);
  }
  public int loopSize(Node node) {
    Node nnn = node;
    return find(nnn, 1);
  }
}

File: d0708/whatever.py
def loop_size(node):
    def find(node, dep):
        setattr(node, 'dfn', dep)
        if hasattr(node.next, 'dfn'):
            return node.dfn - node.next.dfn + 1
        return find(node.next, dep + 1)
    return find(node, 1)